Modules/_asynciomodule.c

```diff
@@ -434,15 +434,33 @@ future_schedule_callbacks(asyncio_state *state, FutureObj *fut)
         return 0;
     }
 
+    if (!PyList_Check(fut->fut_callbacks)) {
```

Comment:
Let's use a faster version since we're controlling this object fully, right?

```suggestion
    if (!PyList_CheckExact(fut->fut_callbacks)) {
```